includes/wikia/models/DesignSystemGlobalFooterModel.class.php

```diff
@@ -0,0 +1,428 @@
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'wikia-community-apps-link-app-store'
+					],
+					'href' => 'https://itunes.apple.com/pl/developer/wikia-inc./id422467077'
+				],
+				[
+					'type' => 'link-image',
+					'image' => 'company/store-googleplay',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'wikia-community-apps-link-google-play'
+					],
+					'href' => 'https://play.google.com/store/apps/developer?id=Wikia,+Inc.'
+				]
+			]
+		],
+		'advertise' => [
+			'header' => [
+				'type' => 'line-text',
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'wikia-advertise-header'
+				]
+			],
+			'links' => [
+				[
+					'type' => 'link-text',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'wikia-advertise-link-media-kit'
+					],
+					'href' => 'http://www.wikia.com/mediakit'
+				],
+				[
+					'type' => 'link-text',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'wikia-advertise-link-contact'
+					],
+					'href' => 'http://www.wikia.com/mediakit#contacts'
+				]
+			]
+		],
+	];
+
+	private $verticalsWithLink = [
+		'games',
+		'movies',
+		'tv',
+	];
+
+	private $wikiId;
+
+	public function __construct( $wikiId ) {
+		parent::__construct();
+
+		$this->wikiId = $wikiId;
+	}
+
+	public function getData() {
+		$this->setLicensingAndVertical();
+
+		return $this->data;
+	}
+
+	private function setLicensingAndVertical() {
+		$params = [
+			'license' => $this->getLicenseData(),
+		];
+		$verticalData = $this->getVerticalData();
+
+		if ( !empty( $verticalData ) ) {
+			$params['vertical'] = $verticalData;
+		}
+
+		$this->data['licensing_and_vertical'] = [
+			'description' => [
+				'type' => 'translatable-text',
+				'key' => 'wikia-licensing-and-vertical-description',
+				'params' => $params
+			],
+		];
+	}
+
+	private function getVerticalData() {
+		$wikiVerticalData = WikiFactoryHub::getInstance()->getWikiVertical( $this->wikiId );
+		$vertical = null;
+		if ( $wikiVerticalData['id'] ) {
+			$vertical = $wikiVerticalData['short'];
+		}
+
+		if ( is_null( $vertical ) ) {
+			return [];
+		}
+
+		$data = [
+			'type' => 'line-text',
+			'title' => [
+				'type' => 'translatable-text',
+				'key' => 'fandom-link-vertical-' . $vertical
+			],
+		];
+
+		if ( in_array( $vertical, $this->verticalsWithLink ) ) {
+			$data['type'] = 'link-text';
+			$data['href'] = 'http://fandom.wikia.com/' . $vertical;
+		}
+
+		return $data;
+	}
+
+	private function getLicenseData() {
+		$licenseText = WikiFactory::getVarByName( 'wgRightsText', $this->wikiId )->cv_value ?: $this->wg->RightsText;
+
+		return [
+			'type' => 'link-text',
+			'title' => [
+				'type' => 'text',
+				'value' => $licenseText
+			],
+			'href' => $this->getLicenseUrl( $this->wikiId )
+		];
+	}
+
+	private function getLicenseUrl() {
+		$licenseUrl = WikiFactory::getVarByName( 'wgRightsUrl', $this->wikiId )->cv_value ?: $this->wg->RightsUrl;
```

Comment:
shouldn't that be the other way round?
read it from memory and if not there possibly go to DB?